File: AlgorithmStudyPlan/HouseRobber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        memoList = [0,0]
     
        for i in range(len(nums)):
            aux = memoList[0]
            memoList[0] = max(memoList[1] + nums[i], memoList[0])
            memoList[1] = aux
        return memoList[0]

File: Problems/HouseRobber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var rob = function(nums) {
    if (!nums.length)
        return 0;
    if (nums.length === 1)
        return nums[0];
    if (nums.length === 2)
        return Math.max(nums[0], nums[1]);
    
    let maxTwoBefore = nums[0];
    let maxOneBefore = Math.max(nums[0], nums[1]);
    
    for (let i = 2; i < nums.length; i++) {
        const maxCurrent = Math.max(nums[i] + maxTwoBefore, maxOneBefore);
        
        maxTwoBefore = maxOneBefore;
        maxOneBefore = maxCurrent;
    }
    
    return maxOneBefore;

};
